src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/DebugPropertyPage/ActiveLaunchProfileExtensionValueProvider.cs

```diff
@@ -0,0 +1,180 @@
+        private readonly IProjectThreadingService _projectThreadingService;
+
+        [ImportingConstructor]
+        public ActiveLaunchProfileExtensionValueProvider(UnconfiguredProject project, ILaunchSettingsProvider launchSettingsProvider, IProjectThreadingService projectThreadingService)
+        {
+            _project = project;
+            _launchSettingsProvider = launchSettingsProvider;
+            _projectThreadingService = projectThreadingService;
+        }
+
+        public override Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            return GetPropertyValueAsync(propertyName);
+        }
+
+        public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            return GetPropertyValueAsync(propertyName);
+        }
+
+        private async Task<string> GetPropertyValueAsync(string propertyName)
+        {
+            ILaunchSettings launchSettings = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
+            ILaunchProfile? activeProfile = launchSettings.ActiveProfile;
+
+            string? activeProfilePropertyValue = propertyName switch
+            {
+                AuthenticationModePropertyName => GetOtherProperty(activeProfile, LaunchProfileExtensions.RemoteAuthenticationModeProperty, string.Empty),
+                NativeDebuggingPropertyName => GetOtherProperty(activeProfile, LaunchProfileExtensions.NativeDebuggingProperty, false) ? True : False,
+                RemoteDebugEnabledPropertyName => GetOtherProperty(activeProfile, LaunchProfileExtensions.RemoteDebugEnabledProperty, false) ? True : False,
+                RemoteDebugMachinePropertyName => GetOtherProperty(activeProfile, LaunchProfileExtensions.RemoteDebugMachineProperty, string.Empty),
+                SqlDebuggingPropertyName => GetOtherProperty(activeProfile, LaunchProfileExtensions.SqlDebuggingProperty, false) ? True : False,
+
+                _ => throw new InvalidOperationException($"{nameof(ActiveLaunchProfileExtensionValueProvider)} does not handle property '{propertyName}'.")
+            };
+
+            return activeProfilePropertyValue ?? string.Empty;
+        }
+
+        public override Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+        {
+            _projectThreadingService.RunAndForget(async () =>
+            {
+                ILaunchSettings launchSettings = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
+
+                var writableLaunchSettings = launchSettings.ToWritableLaunchSettings();
+                var activeProfile = writableLaunchSettings.ActiveProfile;
```

Comment:
Is it possible for "Active" to change before we get to this line? Perhaps it would be better to capture the name of the project in question outside of the `RunAndForget` and update the specific profile?

Reply:
There may we ways to reduce the window of time where that could potentially occur, but to guarantee that we never hit that situation we would need to ensure all paths that read or write the active profile gain exclusive access to it. We should keep that in mind, but right now I think it's unlikely to happen and we can leave things as they are.